app/src/main/java/com/example/flyman3046/allnews/DetailActivity.java

```diff
@@ -0,0 +1,97 @@
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.Toolbar;
+import android.util.Log;
+import android.view.MenuItem;
+import android.widget.TextView;
+
+import com.example.flyman3046.allnews.Model.DataConstants;
+
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.select.Elements;
+
+import java.io.IOException;
+
+/**
+ *
+ */
+public class DetailActivity extends AppCompatActivity {
+    private static final String TAG = DetailActivity.class.getSimpleName();
+
+    private ProgressDialog mProgressDialog;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.article_detail);
+
+        Toolbar myToolbar = (Toolbar) findViewById(R.id.article_detail_toolbar);
+        setSupportActionBar(myToolbar);
+        if(getSupportActionBar() != null) {
+            getSupportActionBar().setDisplayShowTitleEnabled(false);
+            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+        }
+
+        //Need to prevent memory leak with weak reference
+        String url = getIntent().getStringExtra(DataConstants.ARTICLE_LINK_MESSAGE);
+        Log.wtf(TAG, url);
+        new ContentCrawl().execute(url);
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case android.R.id.home:
+                onBackPressed();
+                return true;
+            default:
+                // If we got here, the user's action was not recognized.
+                // Invoke the superclass to handle it.
+                return super.onOptionsItemSelected(item);
+        }
+    }
+
+    // Title AsyncTask
+    private class ContentCrawl extends AsyncTask<String, Void, Void> {
+        String title;
+        Elements element;
+
+        @Override
+        protected void onPreExecute() {
+            super.onPreExecute();
+            mProgressDialog = new ProgressDialog(DetailActivity.this);
+            mProgressDialog.setTitle("Android Basic JSoup Tutorial");
+            mProgressDialog.setMessage("Loading...");
+            mProgressDialog.setIndeterminate(false);
+            mProgressDialog.show();
+        }
+
+        @Override
+        protected Void doInBackground(String... params) {
+            try {
+                // Connect to the web site
+                Document document = Jsoup.connect(params[0]).get();
+                // Get the html document title
+
+                element = document
+                        .select("body div[class=article-entry text]");
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+            return null;
+        }
+
+        @Override
+        protected void onPostExecute(Void result) {
+            // Set title into TextView
+            TextView txttitle = (TextView) findViewById(R.id.test_content);
```

Comment:
keep it as optional, since main purpose is to learn instead of creating a maintainable app.